In [1]:
#IMPORT LIBRARIES
import pandas as pd
import numpy as np
from mlxtend.frequent_patterns import apriori
from mlxtend.frequent_patterns import association_rules

In [2]:
#READ CSV FILE
df = pd.read_excel('OnlineRetail.xlsx', sheet_name='Online Retail')

In [3]:
#VISUALIZE THE CSV
df.head()

InvoiceNo StockCode                          Description  Quantity  \
0    536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1    536365     71053                  WHITE METAL LANTERN         6   
2    536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3    536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4    536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   

          InvoiceDate  UnitPrice  CustomerID         Country  
0 2010-12-01 08:26:00       2.55     17850.0  United Kingdom  
1 2010-12-01 08:26:00       3.39     17850.0  United Kingdom  
2 2010-12-01 08:26:00       2.75     17850.0  United Kingdom  
3 2010-12-01 08:26:00       3.39     17850.0  United Kingdom  
4 2010-12-01 08:26:00       3.39     17850.0  United Kingdom

In [4]:
#DATA EXPLORATION
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 541909 entries, 0 to 541908
Data columns (total 8 columns):
 #   Column       Non-Null Count   Dtype         
---  ------       --------------   -----         
 0   InvoiceNo    541909 non-null  object        
 1   StockCode    541909 non-null  object        
 2   Description  540455 non-null  object        
 3   Quantity     541909 non-null  int64         
 4   InvoiceDate  541909 non-null  datetime64[ns]
 5   UnitPrice    541909 non-null  float64       
 6   CustomerID   406829 non-null  float64       
 7   Country      541909 non-null  object        
dtypes: datetime64[ns](1), float64(2), int64(1), object(4)
memory usage: 33.1+ MB


In [5]:
#DATA EXPLORATION
df.isnull().sum()

InvoiceNo           0
StockCode           0
Description      1454
Quantity            0
InvoiceDate         0
UnitPrice           0
CustomerID     135080
Country             0
dtype: int64

In [6]:
#Visualize the null observations 
df[df['Description'].isnull()]


InvoiceNo StockCode Description  Quantity         InvoiceDate  \
622       536414     22139         NaN        56 2010-12-01 11:52:00   
1970      536545     21134         NaN         1 2010-12-01 14:32:00   
1971      536546     22145         NaN         1 2010-12-01 14:33:00   
1972      536547     37509         NaN         1 2010-12-01 14:33:00   
1987      536549    85226A         NaN         1 2010-12-01 14:34:00   
...          ...       ...         ...       ...                 ...   
535322    581199     84581         NaN        -2 2011-12-07 18:26:00   
535326    581203     23406         NaN        15 2011-12-07 18:31:00   
535332    581209     21620         NaN         6 2011-12-07 18:35:00   
536981    581234     72817         NaN        27 2011-12-08 10:33:00   
538554    581408     85175         NaN        20 2011-12-08 14:06:00   

        UnitPrice  CustomerID         Country  
622           0.0         NaN  United Kingdom  
1970          0.0         NaN  United Kingdom  
1971          0.0         NaN  United Kingdom  
1972          0.0         NaN  United Kingdom  
1987          0.0         NaN  United Kingdom  
...           ...         ...             ...  
535322        0.0         NaN  United Kingdom  
535326        0.0         NaN  United Kingdom  
535332        0.0         NaN  United Kingdom  
536981        0.0         NaN  United Kingdom  
538554        0.0         NaN  United Kingdom  

[1454 rows x 8 columns]

In [7]:
# Deleting the NULL values
df = df.dropna(subset = ['Description'])

#Visualize column again 
df[df['Description'].isnull()]

Empty DataFrame
Columns: [InvoiceNo, StockCode, Description, Quantity, InvoiceDate, UnitPrice, CustomerID, Country]
Index: []

In [8]:
df['InvoiceNo'] = df['InvoiceNo'].astype('str')
df = df[~df['InvoiceNo'].str.contains('C')]

In [9]:
df.Country.value_counts().head(5)


United Kingdom    486168
Germany             9042
France              8408
EIRE                7894
Spain               2485
Name: Country, dtype: int64

In [10]:
df['Description'] = df['Description'].str.strip()

#some of transaction quantity is negative which can not be possible remove that.
df = df[df.Quantity >0]

In [11]:
pd.get_dummies(df, columns=["Description"]).head()

InvoiceNo StockCode  Quantity         InvoiceDate  UnitPrice  CustomerID  \
0    536365    85123A         6 2010-12-01 08:26:00       2.55     17850.0   
1    536365     71053         6 2010-12-01 08:26:00       3.39     17850.0   
2    536365    84406B         8 2010-12-01 08:26:00       2.75     17850.0   
3    536365    84029G         6 2010-12-01 08:26:00       3.39     17850.0   
4    536365    84029E         6 2010-12-01 08:26:00       3.39     17850.0   

          Country  Description_*Boombox Ipod Classic  \
0  United Kingdom                                  0   
1  United Kingdom                                  0   
2  United Kingdom                                  0   
3  United Kingdom                                  0   
4  United Kingdom                                  0   

   Description_*USB Office Mirror Ball  Description_10 COLOUR SPACEBOY PEN  \
0                                    0                                   0   
1                                    0                                   0   
2                                    0                                   0   
3                                    0                                   0   
4                                    0                                   0   

   ...  Description_returned  Description_taig adjust  Description_test  \
0  ...                     0                        0                 0   
1  ...                     0                        0                 0   
2  ...                     0                        0                 0   
3  ...                     0                        0                 0   
4  ...                     0                        0                 0   

   Description_to push order througha s stock was  Description_website fixed  \
0                                               0                          0   
1                                               0                          0   
2                                               0                          0   
3                                               0                          0   
4                                               0                          0   

   Description_wrongly coded 20713  Description_wrongly coded 23343  \
0                                0                                0   
1                                0                                0   
2                                0                                0   
3                                0                                0   
4                                0                                0   

   Description_wrongly marked  Description_wrongly marked 23343  \
0                           0                                 0   
1                           0                                 0   
2                           0                                 0   
3                           0                                 0   
4                           0                                 0   

   Description_wrongly sold (22719) barcode  
0                                         0  
1                                         0  
2                                         0  
3                                         0  
4                                         0  

[5 rows x 4072 columns]

In [12]:
basket = (df[df['Country'] =="France"]
          .groupby(['InvoiceNo', 'Description'])['Quantity']
          .sum().unstack().reset_index().fillna(0)
          .set_index('InvoiceNo'))

In [13]:
basket.head(10)

Description  10 COLOUR SPACEBOY PEN  12 COLOURED PARTY BALLOONS  \
InvoiceNo                                                         
536370                          0.0                         0.0   
536852                          0.0                         0.0   
536974                          0.0                         0.0   
537065                          0.0                         0.0   
537463                          0.0                         0.0   
537468                         24.0                         0.0   
537693                          0.0                         0.0   
537897                          0.0                         0.0   
537967                          0.0                         0.0   
538008                          0.0                         0.0   

Description  12 EGG HOUSE PAINTED WOOD  12 MESSAGE CARDS WITH ENVELOPES  \
InvoiceNo                                                                 
536370                             0.0                              0.0   
536852                             0.0                              0.0   
536974                             0.0                              0.0   
537065                             0.0                              0.0   
537463                             0.0                              0.0   
537468                             0.0                              0.0   
537693                             0.0                              0.0   
537897                             0.0                              0.0   
537967                             0.0                              0.0   
538008                             0.0                              0.0   

Description  12 PENCIL SMALL TUBE WOODLAND  \
InvoiceNo                                    
536370                                 0.0   
536852                                 0.0   
536974                                 0.0   
537065                                 0.0   
537463                                 0.0   
537468                                 0.0   
537693                                 0.0   
537897                                 0.0   
537967                                 0.0   
538008                                 0.0   

Description  12 PENCILS SMALL TUBE RED RETROSPOT  12 PENCILS SMALL TUBE SKULL  \
InvoiceNo                                                                       
536370                                       0.0                          0.0   
536852                                       0.0                          0.0   
536974                                       0.0                          0.0   
537065                                       0.0                          0.0   
537463                                       0.0                          0.0   
537468                                       0.0                          0.0   
537693                                       0.0                          0.0   
537897                                       0.0                          0.0   
537967                                       0.0                          0.0   
538008                                       0.0                          0.0   

Description  12 PENCILS TALL TUBE POSY  12 PENCILS TALL TUBE RED RETROSPOT  \
InvoiceNo                                                                    
536370                             0.0                                 0.0   
536852                             0.0                                 0.0   
536974                             0.0                                 0.0   
537065                             0.0                                 0.0   
537463                             0.0                                 0.0   
537468                             0.0                                 0.0   
537693                             0.0                                 0.0   
537897                             0.0                            

In [14]:
basket['10 COLOUR SPACEBOY PEN'].head(10)

InvoiceNo
536370     0.0
536852     0.0
536974     0.0
537065     0.0
537463     0.0
537468    24.0
537693     0.0
537897     0.0
537967     0.0
538008     0.0
Name: 10 COLOUR SPACEBOY PEN, dtype: float64

In [15]:
def encode_to_binary(x):
    if x > 0:
        return 1
    else: 
        return 0
    

basket_sets = basket.applymap(encode_to_binary)

In [16]:
basket_sets['10 COLOUR SPACEBOY PEN'].head(10)

InvoiceNo
536370    0
536852    0
536974    0
537065    0
537463    0
537468    1
537693    0
537897    0
537967    0
538008    0
Name: 10 COLOUR SPACEBOY PEN, dtype: int64

In [17]:
#remove postage item as it is just a seal which almost all transaction contain. 
basket_sets.drop(columns=['POSTAGE'],inplace=True)

In [18]:
df.InvoiceNo.value_counts()

573585    1114
581219     749
581492     731
580729     721
558475     705
          ... 
561811       1
557754       1
574014       1
557002       1
540251       1
Name: InvoiceNo, Length: 20136, dtype: int64

In [19]:
#call apriori function and pass minimum support here we are passing 7%. 
#means 7 times in total number of transaction that item was present.
frequent_itemsets = apriori(basket_sets, min_support=0.07, use_colnames=True)

In [20]:
frequent_itemsets

support                                           itemsets
0   0.071429                      (4 TRADITIONAL SPINNING TOPS)
1   0.096939                       (ALARM CLOCK BAKELIKE GREEN)
2   0.102041                        (ALARM CLOCK BAKELIKE PINK)
3   0.094388                         (ALARM CLOCK BAKELIKE RED)
4   0.081633                     (BAKING SET 9 PIECE RETROSPOT)
5   0.071429                     (CHILDRENS CUTLERY DOLLY GIRL)
6   0.099490                             (DOLLY GIRL LUNCH BOX)
7   0.096939                          (JUMBO BAG RED RETROSPOT)
8   0.076531                       (JUMBO BAG WOODLAND ANIMALS)
9   0.125000                           (LUNCH BAG APPLE DESIGN)
10  0.084184                      (LUNCH BAG DOLLY GIRL DESIGN)
11  0.153061                          (LUNCH BAG RED RETROSPOT)
12  0.119898                        (LUNCH BAG SPACEBOY DESIGN)
13  0.117347                               (LUNCH BAG WOODLAND)
14  0.142857                 (LUNCH BOX WITH CUTLERY RETROSPOT)
15  0.104592                           (MINI PAINT SET VINTAGE)
16  0.102041                  (PACK OF 72 RETROSPOT CAKE CASES)
17  0.081633                          (PAPER BUNTING RETROSPOT)
18  0.168367                    (PLASTERS IN TIN CIRCUS PARADE)
19  0.137755                         (PLASTERS IN TIN SPACEBOY)
20  0.081633                        (PLASTERS IN TIN STRONGMAN)
21  0.170918                 (PLASTERS IN TIN WOODLAND ANIMALS)
22  0.188776                               (RABBIT NIGHT LIGHT)
23  0.096939                      (RED RETROSPOT CHARLOTTE BAG)
24  0.137755                         (RED RETROSPOT MINI CASES)
25  0.071429                         (RED RETROSPOT PICNIC BAG)
26  0.181122                    (RED TOADSTOOL LED NIGHT LIGHT)
27  0.125000                         (REGENCY CAKESTAND 3 TIER)
28  0.086735                  (RETROSPOT TEA SET CERAMIC 11 PC)
29  0.107143                (ROUND SNACK BOXES SET OF 4 FRUITS)
30  0.158163               (ROUND SNACK BOXES SET OF4 WOODLAND)
31  0.076531                (SET/10 RED POLKADOT PARTY CANDLES)
32  0.132653               (SET/20 RED RETROSPOT PAPER NAPKINS)
33  0.137755                      (SET/6 RED SPOTTY PAPER CUPS)
34  0.127551                    (SET/6 RED SPOTTY PAPER PLATES)
35  0.071429                           (SPACEBOY BIRTHDAY CARD)
36  0.125000                               (SPACEBOY LUNCH BOX)
37  0.122449                (STRAWBERRY LUNCH BOX WITH CUTLERY)
38  0.094388                          (TEA PARTY BIRTHDAY CARD)
39  0.073980                           (WOODLAND CHARLOTTE BAG)
40  0.073980  (ALARM CLOCK BAKELIKE PINK, ALARM CLOCK BAKELI...
41  0.079082  (ALARM CLOCK BAKELIKE RED, ALARM CLOCK BAKELIK...
42  0.073980  (ALARM CLOCK BAKELIKE PINK, ALARM CLOCK BAKELI...
43  0.071429         (SPACEBOY LUNCH BOX, DOLLY GIRL LUNCH BOX)
44  0.089286  (PLASTERS IN TIN CIRCUS PARADE, PLASTERS IN TI...
45  0.102041  (PLASTERS IN TIN CIRCUS PARADE, PLASTERS IN TI...
46  0.104592  (PLASTERS IN TIN WOODLAND ANIMALS, PLASTERS IN...
47  0.102041  (SET/20 RED RETROSPOT PAPER NAPKINS, SET/6 RED...
48  0.102041  (SET/6 RED SPOTTY PAPER PLATES, SET/20 RED RET...
49  0.122449  (SET/6 RED SPOTTY PAPER PLATES, SET/6 RED SPOT...
50  0.099490  (SET/6 RED SPOTTY PAPER PLATES, SET/20 RED RET...

In [21]:
rules = association_rules(frequent_itemsets, metric="lift", min_threshold=1)
rules.head()

antecedents                   consequents  \
0   (ALARM CLOCK BAKELIKE PINK)  (ALARM CLOCK BAKELIKE GREEN)   
1  (ALARM CLOCK BAKELIKE GREEN)   (ALARM CLOCK BAKELIKE PINK)   
2    (ALARM CLOCK BAKELIKE RED)  (ALARM CLOCK BAKELIKE GREEN)   
3  (ALARM CLOCK BAKELIKE GREEN)    (ALARM CLOCK BAKELIKE RED)   
4   (ALARM CLOCK BAKELIKE PINK)    (ALARM CLOCK BAKELIKE RED)   

   antecedent support  consequent support   support  confidence      lift  \
0            0.102041            0.096939  0.073980    0.725000  7.478947   
1            0.096939            0.102041  0.073980    0.763158  7.478947   
2            0.094388            0.096939  0.079082    0.837838  8.642959   
3            0.096939            0.094388  0.079082    0.815789  8.642959   
4            0.102041            0.094388  0.073980    0.725000  7.681081   

   leverage  conviction  
0  0.064088    3.283859  
1  0.064088    3.791383  
2  0.069932    5.568878  
3  0.069932    4.916181  
4  0.064348    3.293135

In [22]:
rules[ (rules['lift'] >= 8) &
       (rules['confidence'] >= 0.6) ]

antecedents                   consequents  \
2    (ALARM CLOCK BAKELIKE RED)  (ALARM CLOCK BAKELIKE GREEN)   
3  (ALARM CLOCK BAKELIKE GREEN)    (ALARM CLOCK BAKELIKE RED)   

   antecedent support  consequent support   support  confidence      lift  \
2            0.094388            0.096939  0.079082    0.837838  8.642959   
3            0.096939            0.094388  0.079082    0.815789  8.642959   

   leverage  conviction  
2  0.069932    5.568878  
3  0.069932    4.916181

In [23]:
basket['ALARM CLOCK BAKELIKE RED'].sum()


316.0

In [24]:
basket['ALARM CLOCK BAKELIKE GREEN'].sum()

340.0

The highest lift value of 8 was between the Alarm Clock Bakelike Red and Alarm Clock Bakelike Green. When we explored further, 340 people bought the two items meaning all the people who bought the red clock bought the green clock too.  

In [25]:
rules[ (rules['lift'] >= 5) &
       (rules['confidence'] >= 0.6) ]

antecedents  \
0                         (ALARM CLOCK BAKELIKE PINK)   
1                        (ALARM CLOCK BAKELIKE GREEN)   
2                          (ALARM CLOCK BAKELIKE RED)   
3                        (ALARM CLOCK BAKELIKE GREEN)   
4                         (ALARM CLOCK BAKELIKE PINK)   
5                          (ALARM CLOCK BAKELIKE RED)   
7                              (DOLLY GIRL LUNCH BOX)   
14               (SET/20 RED RETROSPOT PAPER NAPKINS)   
15                      (SET/6 RED SPOTTY PAPER CUPS)   
16                    (SET/6 RED SPOTTY PAPER PLATES)   
17               (SET/20 RED RETROSPOT PAPER NAPKINS)   
18                    (SET/6 RED SPOTTY PAPER PLATES)   
19                      (SET/6 RED SPOTTY PAPER CUPS)   
20  (SET/6 RED SPOTTY PAPER PLATES, SET/20 RED RET...   
21  (SET/6 RED SPOTTY PAPER PLATES, SET/6 RED SPOT...   
22  (SET/20 RED RETROSPOT PAPER NAPKINS, SET/6 RED...   
23                    (SET/6 RED SPOTTY PAPER PLATES)   
24               (SET/20 RED RETROSPOT PAPER NAPKINS)   
25                      (SET/6 RED SPOTTY PAPER CUPS)   

                                          consequents  antecedent support  \
0                        (ALARM CLOCK BAKELIKE GREEN)            0.102041   
1                         (ALARM CLOCK BAKELIKE PINK)            0.096939   
2                        (ALARM CLOCK BAKELIKE GREEN)            0.094388   
3                          (ALARM CLOCK BAKELIKE RED)            0.096939   
4                          (ALARM CLOCK BAKELIKE RED)            0.102041   
5                         (ALARM CLOCK BAKELIKE PINK)            0.094388   
7                                (SPACEBOY LUNCH BOX)            0.099490   
14                      (SET/6 RED SPOTTY PAPER CUPS)            0.132653   
15               (SET/20 RED RETROSPOT PAPER NAPKINS)            0.137755   
16               (SET/20 RED RETROSPOT PAPER NAPKINS)            0.127551   
17                    (SET/6 RED SPOTTY PAPER PLATES)            0.132653   
18                      (SET/6 RED SPOTTY PAPER CUPS)            0.127551   
19                    (SET/6 RED SPOTTY PAPER PLATES)            0.137755   
20                      (SET/6 RED SPOTTY PAPER CUPS)            0.102041   
21               (SET/20 RED RETROSPOT PAPER NAPKINS)            0.122449   
22                    (SET/6 RED SPOTTY PAPER PLATES)            0.102041   
23  (SET/20 RED RETROSPOT PAPER NAPKINS, SET/6 RED...            0.127551   
24  (SET/6 RED SPOTTY PAPER PLATES, SET/6 RED SPOT...            0.132653   
25  (SET/6 RED SPOTTY PAPER PLATES, SET/20 RED RET...            0.137755   

    consequent support   support  confidence      lift  leverage  conviction  
0             0.096939  0.073980    0.725000  7.478947  0.064088    3.283859  
1             0.102041  0.073980    0.763158  7.478947  0.064088    3.791383  
2             0.096939  0.079082    0.837838  8.642959  0.069932    5.568878  
3             0.094388  0.079082    0.815789  8.642959  0.069932    4.916181  
4             0.094388  0.073980    0.725000  7.681081  0.064348    3.293135  
5             0.102041  0.073980    0.783784  7.681081  0.064348    4.153061  
7             0.125000  0.071429    0.717949  5.743590  0.058992    3.102273  
14            0.137755  0.102041    0.769231  5.584046  0.083767    3.736395  
15            0.132653  0.102041    0.740741  5.584046  0.083767    3.345481  
16            0.132653  0.102041    0.800000  6.030769  0.085121    4.336735  
17            0.127551  0.102041    0.769231  6.030769  0.085121    3.780612  
18            0.137755  0.122449    0.960000  6.968889  0.104878   21.556122  
19            0.127551  0.122449    0.888889  6.968889  0.104878    7.852041  
20            0.137755  0.099490    0.975000  7.077778  0.085433   34.489796  
21            0.132653  0.099490    0.812500  6.125000  0.083247    4.625850  
22            0.127551  0.099490    0.975000  7.644000  0.086474   34.897959  
23            0.1

In [28]:
basket2 = (df[df['Country'] =="Germany"]
          .groupby(['InvoiceNo', 'Description'])['Quantity']
          .sum().unstack().reset_index().fillna(0)
          .set_index('InvoiceNo'))

basket_sets = basket.applymap(encode_to_binary)
basket_sets.drop('POSTAGE', inplace=True, axis=1)
basket_sets2 = basket2.applymap(encode_to_binary)
basket_sets2.drop('POSTAGE', inplace=True, axis=1)
frequent_itemsets2 = apriori(basket_sets2, min_support=0.05, use_colnames=True)
rules2 = association_rules(frequent_itemsets2, metric="lift", min_threshold=1)

rules2[ (rules2['lift'] >= 4) &
        (rules2['confidence'] >= 0.5)]

antecedents                         consequents  \
0   (PLASTERS IN TIN CIRCUS PARADE)  (PLASTERS IN TIN WOODLAND ANIMALS)   
7        (PLASTERS IN TIN SPACEBOY)  (PLASTERS IN TIN WOODLAND ANIMALS)   
11    (RED RETROSPOT CHARLOTTE BAG)            (WOODLAND CHARLOTTE BAG)   

    antecedent support  consequent support   support  confidence      lift  \
0             0.115974            0.137856  0.067834    0.584906  4.242887   
7             0.107221            0.137856  0.061269    0.571429  4.145125   
11            0.070022            0.126915  0.059081    0.843750  6.648168   

    leverage  conviction  
0   0.051846    2.076984  
7   0.046488    2.011670  
11  0.050194    5.587746

In [29]:
basket['RED RETROSPOT CHARLOTTE BAG'].sum()

469.0

In [30]:
basket['WOODLAND CHARLOTTE BAG'].sum()

305.0

On the other hand, in Germany, there were only three rules that had lift value 4 and above. These the rules represent the strongest of all the rules in the Germany dataset. The highest value is of 6 and is between the RED RETROSPOT CHARLOTTE BAG and WOODLAND CHARLOTTE BAG. 469 people bought the former and 305 bought the latter. This implies that 305 out of the 469 people that bought the Red Retro bag also bought the woodland bag. 

In [31]:
basket3 = (df[df['Country'] =="Spain"]
          .groupby(['InvoiceNo', 'Description'])['Quantity']
          .sum().unstack().reset_index().fillna(0)
          .set_index('InvoiceNo'))

basket_sets3 = basket3.applymap(encode_to_binary)
frequent_itemsets3 = apriori(basket_sets3, min_support=0.06, use_colnames=True)
rules3 = association_rules(frequent_itemsets3, metric="lift", min_threshold=1)

rules3[ (rules3['lift'] >= 11) &
        (rules3['confidence'] >= 0.5)]

antecedents  \
48            (POPPY'S PLAYHOUSE KITCHEN)   
49            (POPPY'S PLAYHOUSE BEDROOM)   
70   (SET/10 PINK POLKADOT PARTY CANDLES)   
71   (SET OF 6 GIRLS CELEBRATION CANDLES)   
103  (POPPY'S PLAYHOUSE KITCHEN, POSTAGE)   
104  (POPPY'S PLAYHOUSE BEDROOM, POSTAGE)   
105           (POPPY'S PLAYHOUSE KITCHEN)   
106           (POPPY'S PLAYHOUSE BEDROOM)   

                              consequents  antecedent support  \
48            (POPPY'S PLAYHOUSE BEDROOM)            0.077778   
49            (POPPY'S PLAYHOUSE KITCHEN)            0.088889   
70   (SET OF 6 GIRLS CELEBRATION CANDLES)            0.077778   
71   (SET/10 PINK POLKADOT PARTY CANDLES)            0.077778   
103           (POPPY'S PLAYHOUSE BEDROOM)            0.066667   
104           (POPPY'S PLAYHOUSE KITCHEN)            0.077778   
105  (POPPY'S PLAYHOUSE BEDROOM, POSTAGE)            0.077778   
106  (POPPY'S PLAYHOUSE KITCHEN, POSTAGE)            0.088889   

     consequent support   support  confidence       lift  leverage  conviction  
48             0.088889  0.077778    1.000000  11.250000  0.070864         inf  
49             0.077778  0.077778    0.875000  11.250000  0.070864    7.377778  
70             0.077778  0.066667    0.857143  11.020408  0.060617    6.455556  
71             0.077778  0.066667    0.857143  11.020408  0.060617    6.455556  
103            0.088889  0.066667    1.000000  11.250000  0.060741         inf  
104            0.077778  0.066667    0.857143  11.020408  0.060617    6.455556  
105            0.077778  0.066667    0.857143  11.020408  0.060617    6.455556  
106            0.066667  0.066667    0.750000  11.250000  0.060741    3.733333

Meanwhile, in Spain, there are 8 rules with a lift of 11 and two out of the eight has a confidence of 1.0 which happens to be between POPPY'S PLAYHOUSE KITCHEN and POPPY'S PLAYHOUSE BEDROOM. This appears to be one of the strongest association rules from the entire dataset. 